packages/data-service/README.md

```diff
@@ -0,0 +1,113 @@
+
+```js
+  /*
+   * Receives a stream of React.Component and outputs a single components that 
+   * renders the most recevent event from the Observable.
+  */
+  renderComponents(component$: Observable<React.Component>): React.Component;
+```
+
+Here is an example of how it could be used to render a page for `space shuttles`
+that shows which shuttles are on space, and which are ready to be launched.
+
+```js
+// Imagine we manage a launchpad for interplanetary ships
+import { shuttles$, launchShuttle } from "shuttles/streams";
+import { Space, LaunchPad, LoadingUI, ErrorMessageUI } from "ui/components"
+import { renderComponents } from "data-service"
+
+
+// Merging this streams should be done via graphql in the future
+// See bellow for the equivalent graphql.
+const launched$ = shuttles$.filter((sht) => sht.location === "space");
+const nextLaunches$ = shuttles$.filter((sht) => sht.location === "earth");
+const content$ = launched$.combineLatest(nextLaunches$,
+                          (launched, nextLaunches) => {launched, nextLaunches});
+
+// This would be a graphql mutation
+const launch = (name) => {
+  launchShuttle({name, launchDate: new Date()})
+}
+
+const components$ = content$.map((data) => 
+                                <World>
+                                  <LaunchPad shuttles={data.nextLaunches} />
+                                  <Space shuttles={data.launched} />
+                                  <LaunchButton onLaunch={launch} />
+                                </World>)
+                            .startWith(<LoadingUI />)                                
+                            .catch((err) => <ErrorMessageUI err>));
+
+export default renderComponents(components$);
+```
+
+Notice how easy is to provide error handling whenever an exception happens and how 
+we could start the stream of components providing a loading component.
+
+If you would be using graphql, the `content$` would be creted through grpahql and the code
+would look like:
+```js
+// Imagine we manage a launchpad for interplanetary ships
+import { shuttles$, launchShuttle } from "shuttles/streams";
+import { Space, LaunchPad, LoadingUI, ErrorMessageUI } from "ui/components"
+import { renderComponents } from "data-service"
+
+import { shuttleSchema } from "shuttles/schema";
+
+const query = `
+  query launched($location){ 
+    shuttle(location: $location) {
+      name
+      launchDate
+    } 
+  }
+  
+  query nextLaunches($location){ 
+    shuttle(location: $location) {
+      name
+    } 
+  }
+`);
+
+const content$ = graphql(query, shuttleSchema); 
+
+const launchMutation = `
+  mutation {
+    launchShip(name: $name, launchDate: $launchDate) {
+      name
+      launchDate 
+    }
+  }
+`;
+
+// This would be a graphql mutation
+const launch = (name) => {
+  const launchDate = new Date();
+  graphql(launchMutation, shuttleSchema).mutate({ name, launchDate });
+}
+
+const components$ = content$.map((data) => 
```

Comment:
How does the data get passed into the graphql query? I mean $location needs be specified for the query to be resolvable, right?

Reply:
At this point of the writing that is not clear. I still have to figure out the details on how we are going to use graphl which will be the rest of the work I am doing. Notice that the location is hardcoded on the non-graphql example anyway. 

To make this example whole, I can remove `$location`.

Reply:
I solved it in a different way, by adding an idea about how the parameters could be.